pipeline.py

```diff
@@ -280,22 +286,230 @@ def pipeline(
     return pipeline
 
 
-@click.command()
 @click.option(
     "--mock",
     type=click.Choice(MOCKED_STAGES, case_sensitive=False),
     help="Mock part of the pipeline",
     multiple=True,
     default=[],
 )
-def cli(mock):
+@click.group(invoke_without_command=True)
+@click.pass_context
+def cli(ctx: click.Context, mock):
+    if ctx.invoked_subcommand is None:
+        generate_pipeline(mock)
 
+
+def generate_pipeline(mock):
     p = pipeline_wrapper(mock)
 
     with click.progressbar(length=1, label="Generating pipeline") as bar:
-        compiler.Compiler().compile(p, "pipeline.yaml")
+        compiler.Compiler().compile(p, PIPELINE_FILE_NAME)
         bar.update(1)
 
 
+@cli.command(name="gen-standalone")
+def gen_standalone():
+    """
+    Generates a standalone script that mimics the behavior of the pipeline.
+
+    This function should be used when Kubeflow Pipelines are not available. It will generate a
+    script that replicates the pipeline's functionality.
+
+    Example usage: ``` $ python pipeline.py gen-standalone ```
+    """
+    from jinja2 import Template
+    from jinja2.exceptions import TemplateSyntaxError
+    import yaml
+    import json
+
+    click.echo("Generating pipeline YAML file...")
+    try:
+        generate_pipeline(mock=None)
+    except ValueError as e:
+        click.echo(f"Error: {e}", err=True)
+        raise click.exceptions.Exit(1)
+
+    # Load the YAML pipeline file which contains multiple documents
+    with open(PIPELINE_FILE_NAME, "r", encoding="utf-8") as file:
+        try:
+            documents = list(yaml.safe_load_all(file))
+        except ValueError as e:
+            click.echo(f"Error: {e}", err=True)
+            raise click.exceptions.Exit(1)
+
+    # The list of executor names to extract details from to generate the standalone script
+    executor_names = [
+        "exec-data-processing-op",
+        "exec-sdg-op",
+        "exec-git-clone-op",
+    ]
+
+    details = {}
+    for executor_name in executor_names:
+        try:
+            executor_name_camelize = executor_name.replace("-", "_")
+            # replace "-" with "_" in executor_name to match the key in the details dictionary
+            executor_details = get_executor_details(documents, executor_name)
+            if executor_details is not None:
+                details[executor_name_camelize + "_image"] = executor_details["image"]
+                details[executor_name_camelize + "_command"] = executor_details[
+                    "command"
+                ]
+                details[executor_name_camelize + "_args"] = remove_template_markers(
+                    executor_details["args"], executor_name_camelize
+                )
+        except ValueError as e:
+            click.echo(f"Error: {e}", err=True)
+            raise click.exceptions.Exit(1)
+
+    # Populate the KFP model server ConfigMap
+    with open(
+        path.join(path.dirname(__file__), KFP_MODEL_SERVER_CM), encoding="utf-8"
```

Comment:
Do you want to use the `os` package here? I was getting errors locally trying to generate the `standalone.py` without this change. 
```suggestion
        os.path.join(os.path.dirname(__file__), KFP_MODEL_SERVER_CM), encoding="utf-8"
```

Reply:
What was the error?

Reply:
`NameError: name 'path' is not defined`

Reply:
Hum I have this line `from os import path` above so it should work, perhaps I wasn't present before? Have you tried again?